cmplib/cmpcl_read.c

```diff
@@ -1253,8 +1327,8 @@ int cmp_pkimessage_parse_check_der(cmp_ctx *ctx, int expected_type, cmp_pkimessa
         }
 
         if( ( ret = cmp_x509_crt_verify(ctx->extraCerts, //mbedtls_x509_crt *crt,
-                ctx->prot_trust_anchor, //mbedtls_x509_crt *trust_ca,
-                ctx->prot_crls, //mbedtls_x509_crl *ca_crl,
+                ctx->enrol_trust_anchor, //mbedtls_x509_crt *trust_ca,
+                ctx->enrol_crls, //mbedtls_x509_crl *ca_crl,
```

Comment:
Why did you change this?
This looks wrong because at this point the (general) message protection is checked for any kind of message,
not the cert returned in a positive ip/cp/kup. Yet for the latter situation, would be nice to verify the received new cert in addition, using the `enrol_{trust_anchor,enrol_crtls}`, in `cmpcl_do_transaction()`.